client/lib/analytics/ad-tracking/ad-track-trial-start.ts

```diff
@@ -0,0 +1,44 @@
+import { mayWeTrackByTracker } from '../tracker-buckets';
+
+/**
+ * We'll be accessing rdt from the window object. These definitions are homemade and not from Reddit (not documented, as the example script is JS),
+ * but they make the validation happy.
+ */
+declare global {
+	interface Window {
+		rdt: {
+			callQueue: object[];
+			sendEvent: ( ...args: [ string, string?, object? ] ) => void;
+		} & ( ( ...args: [ string, string?, object? ] ) => void );
+	}
+}
+
+/**
+ * Tracks a lead (free trial) in Reddit.
+ */
+export const redditTrackerFreeTrialStarted = ( trial_flow_name: string ): void => {
+	if ( ! mayWeTrackByTracker( 'reddit' ) ) {
+		return;
+	}
+
+	const params = {
+		products: [
+			{
+				id: 'wpcom_creator_trial',
+				name: trial_flow_name,
+				category: 'free_trial',
+			},
+		],
+	};
+	window.rdt && window.rdt( 'track', 'Lead', params );
```

Comment:
Is this for a case where `window` itself is undefined? (it will still throw "uncaught reference error" in that case)

Reply:
Thanks for the review! It's actually just a safeguard in case `rdt` is not set on the window, so we don't try to call the potentially non-existent function. But it's actually redundant, as we're checking `mayWeTrackByTracker('reddit');` and returning early if that's falsy, and that function already calls `isTrackerIntialized` where this check is done (`() => 'rdt' in window`).

So I can remove this check, as it must be true by the time we're at this line. :)

Reply:
@chriskmnds -- Sorted this check now, and also sorted the other review feedback you had in the flow. 👍

Reply:
Thanks for addressing and clarifying :-)